### PR TITLE
test: user completion modal

### PR DESCRIPTION
**Pull request by IrakliJani**
Closes: 

### Description:

- adds jest-dom / react-testing-library eslint plugins/extensions
- improves mocks
- adds component tests for `UserCompletionModal`
- fixes bug in the component where signing up a new org member would enable domain auth
  - ^^^ this is why we need these tests 🤷 

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx [hunk 1]
```diff
@@ -1,24 +1,286 @@
...
+        expect(subscriptionCheckbox).toBeInTheDocument();
+        expect(subscriptionCheckbox).toBeChecked();
+        await user.click(subscriptionCheckbox);
+
+        // check tracking checkbox (unchecked by default)
+        const trackingCheckbox = await screen.findByRole('checkbox', {
+            name: `Anonymize my usage data`,
+        });
+        expect(trackingCheckbox).not.toBeChecked();
+        expect(trackingCheckbox).toBeInTheDocument();
```
Comment (joaoviana) [inline packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx:206]: guess this isn't needed if 205 is true no?

Comment (joaoviana) [reply packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx:206]: or maybe just swap?

File: packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx [hunk 2]
```diff
@@ -1,24 +1,286 @@
...
+                isTrackingAnonymized: true,
+            })
+            .reply(200);
+
+        // submit form
+        await user.click(submitButton);
+
+        // wait for api call to be made
+        scope.done();
+        await waitFor(() => expect(scope.isDone()).toBe(true));
```
Comment (joaoviana) [inline packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx:228]: would it be possible to check if the dialog isn't open anymore?

File: packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx [hunk 3]
```diff
@@ -1,24 +1,286 @@
...
+                isTrackingAnonymized: false,
+            })
+            .reply(200);
+
+        // submit form
+        await user.click(submitButton);
+
+        // wait for api call to be made
+        scope.done();
+        await waitFor(() => expect(scope.isDone()).toBe(true));
```
Comment (joaoviana) [inline packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx:284]: same here perhaps ^ (on asserting if dialog is closed)

**Review by joaoviana — APPROVED** [review]
🙌 

Just some non-blocking comments! 

Looks great 🙏

File: packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx [hunk 4]
```diff
@@ -1,24 +1,286 @@
...
 
     it("should render user completion modal if user's setup is not complete", async () => {
-        const { getByText } = renderWithProviders(<UserCompletionModal />, {
+        renderWithProviders(<UserCompletionModal />, {
+            user: {
+                isSetupComplete: false,
+            },
+        });
+
+        const welcomeModel = await screen.findByRole('dialog', {
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx:23]: hate it when I do this 😂 thanks!

File: packages/frontend/src/testing/testUtils.tsx [hunk 5]
```diff
@@ -27,9 +26,7 @@ const getMockedProviders = (appMocks?: AppProviderMockProps['mocks']) => {
             <ReactQueryProvider>
                 <MantineProvider>
                     <AppProviderMock mocks={appMocks}>
-                        <BrowserRouter>
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/frontend/src/testing/testUtils.tsx:30]: idk, I was stuck somewhere and I was removing random code. I think this is not necessary as we are probably going to mock hooks provided by the router anyways..

File: packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx (hunk 2 above)
Comment (IrakliJani) [reply packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx:228]: good idea. I might do that in a separate PR tho.

File: packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx (hunk 1 above)
Comment (IrakliJani) [reply packages/frontend/src/components/UserCompletionModal/UserCompletionModal.test.tsx:205]: swap it is!

**Pull request merged**